traversal/procedure/ProcedureVertex.java

```diff
@@ -106,18 +102,54 @@ public ProcedureVertex.Type asType() {
         throw TypeDBException.of(ILLEGAL_CAST, className(this.getClass()), className(ProcedureVertex.Type.class));
     }
 
+    public int order() {
+        return order;
+    }
+
+    public List<ProcedureEdge<?, ?>> orderedOuts() {
+        if (orderedOuts == null) {
+            orderedOuts = outs().stream().sorted(Comparator.comparingInt(e -> e.to().order())).collect(toList());
+        }
+        return orderedOuts;
+    }
+
+    public Set<ProcedureVertex<?, ?>> transitiveTos() {
+        if (transitiveOuts == null) {
+            HashSet<ProcedureVertex<?, ?>> transitive = new HashSet<>();
+            outs().forEach(edge -> {
+                transitive.add(edge.to());
+                transitive.addAll(edge.to().transitiveTos());
+            });
+            transitiveOuts = transitive;
+        }
+        return transitiveOuts;
+    }
+
+    public Set<Identifier.Variable> scopesVisited() {
+        if (scopesVisited == null) scopesVisited = computeScopesVisited();
+        return scopesVisited;
+    }
+
+    Set<Identifier.Variable> computeScopesVisited() {
+        return set();
+    }
+
+    void setOrder(int order) {
```

Comment:
This setter needs to be removed this when the query planner natively orders vertices

cc @dmitrii-ubskii